test/php/Nominatim/ParameterParserTest.php

```diff
@@ -0,0 +1,216 @@
+                                        'float3' => '',
+                                        'float4' => 0
+                                       ]);
+
+        $this->assertSame(false, $oParams->getFloat('non-exists'));
+        $this->assertSame(999, $oParams->getFloat('non-exists', 999));
+        $this->assertSame(1.0, $oParams->getFloat('float1'));
+        $this->assertSame(-5.0, $oParams->getFloat('float2'));
+        $this->assertSame(false, $oParams->getFloat('float3')); // FIXME: should be 0 instead?
+        $this->assertSame(0.0, $oParams->getFloat('float4'));
+    }
+
+
+    public function testGetFloatWithString()
+    {
+        $this->setExpectedException(Exception::class, "Floating-point number expected for parameter 'float5'");
+        (new ParameterParser(['float5' => 'a']))->getFloat('float5');
+    }
+
+
+    public function testGetFloatWithInvalidNumber()
+    {
+        $this->setExpectedException(Exception::class, "Floating-point number expected for parameter 'float6'");
+        (new ParameterParser(['float6' => '-55.']))->getFloat('float6');
+    }
+
+
+    public function testGetString()
+    {
+        $oParams = new ParameterParser([
+                                        'str1' => 'abc',
+                                        'str2' => '',
+                                        'str3' => '0'
+                                       ]);
+
+        $this->assertSame(false, $oParams->getString('non-exists'));
+        $this->assertSame('default', $oParams->getString('non-exists', 'default'));
+        $this->assertSame('abc', $oParams->getString('str1'));
+        $this->assertSame(false, $oParams->getStringList('str2'));
+        $this->assertSame(false, $oParams->getStringList('str3')); // FIXME: should be 0 instead?
+    }
+
+
+    public function testGetSet()
+    {
+        $oParams = new ParameterParser([
+                                        'val1' => 'foo',
+                                        'val2' => '',
+                                        'val3' => 0
+                                       ]);
+
+        $this->assertSame(false, $oParams->getSet('non-exists', ['foo', 'bar']));
+        // FIXME: unclear if the default value has to be part of the set
+        $this->assertSame('default', $oParams->getSet('non-exists', ['foo', 'bar'], 'default'));
+        $this->assertSame('foo', $oParams->getSet('val1', ['foo', 'bar']));
+
+        $this->assertSame(false, $oParams->getSet('val2', ['foo', 'bar']));
+        $this->assertSame(0, $oParams->getSet('val3', ['foo', 'bar']));
+    }
+
+
+    public function testGetSetWithValueNotInSet()
+    {
+        $this->setExpectedException(Exception::class, "Parameter 'val4' must be one of: foo, bar");
+        (new ParameterParser(['val4' => 'faz']))->getSet('val4', ['foo', 'bar']);
+    }
+
+
+    public function testGetStringList()
+    {
+        $oParams = new ParameterParser([
+                                        'list1' => ',a,b,c,,c,d',
```

Comment:
How about adding a normal list without empty spots?

Reply:
Just wanted to make sure it's we don't have unexpected behavior. Empty strings are now ignored.